packages/compiler/src/core/logger/console-sink.ts

```diff
@@ -129,3 +132,44 @@ function getLineAndColumn(location: SourceLocation): RealLocation {
   }
   return result;
 }
+
+export async function trackAction<T>(asyncAction: () => Promise<T>, log: ProcessedLog): Promise<T> {
```

Comment:
Sorry to keep going back on comment here, but maybe because this is special the `log` can just be `message: string`

Reply:
and I wouild put message first so it looks cleaner to call

```ts
.trackAction("Running foo", () => {
  ...
});
```